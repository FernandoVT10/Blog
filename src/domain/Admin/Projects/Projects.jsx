import SkillList from "./SkillList";

import EditProject from "./ProjectEditor/EditProject";
import AddProject from "./ProjectEditor/AddProject";

import { useRouter } from "next/router";

const Projects =  () => {
    const router = useRouter();

    const options = router.query.options || [];

    const getModals = () => {
        if(options[0] === "addProject") {
            return <AddProject/>;
        }

        if(options[1] === "edit") {
            return <EditProject projectId={options[0]}/>;
        }
    }

    return (
        <div>
            { getModals() }

            <SkillList/>
        </div>
    );
}

export default Projects;