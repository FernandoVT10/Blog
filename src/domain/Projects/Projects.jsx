import ProjectList from "./ProjectList";
import ViewProject from "./ViewProject";

import ApiController from "@/services/ApiController";

import { useEffect, useState, useRef } from "react";
import { useRouter } from "next/router";

const Projects = () => {
    const [projects, setProjects] = useState([]);
    const [project, setProject] = useState({});
    const [loading, setLoading] = useState(true);

    const currentProjectId = useRef();

    const router = useRouter();

    useEffect(() => {
        ApiController.get("projects")
        .then(res => {
            if(res.data) {
                setProjects(res.data.projects);
            }

            setLoading(false);
        });
    }, []);

    useEffect(() => {
        const projectId = router.query.project;

        if(projectId && projectId !== currentProjectId.current) {
            setLoading(true);
            setProject({});
            currentProjectId.current = projectId;

            ApiController.get(`projects/${projectId}`)
            .then(res => {
                if(res.data) {
                    setProject(res.data.project);
                }
                setLoading(false);
            });
        }
    }, [router.query]);

    if(router.query.project) {
        return <ViewProject project={project} loading={loading} />;
    }

    return <ProjectList projects={projects} loading={loading} />;
}

export default Projects;