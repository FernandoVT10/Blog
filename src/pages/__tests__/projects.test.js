import { create } from "react-test-renderer";
import Projects from "../projects";


const SKILL_MOCKS = [
    {
        _id: 18,
        name: "React JS",
        image: "skill-1.jpg",
        color: "blue"
    },
    {
        _id: 19,
        name: "Node JS",
        image: "skill-2.jpg",
        color: "green"
    },
    {
        _id: 20,
        name: "GraphQl",
        image: "skill-3.jpg",
        color: "purple"
    }
];

describe("Project page", () => {
    it("It check if renders correctly", async () => {
        const component = create(<Projects skills={SKILL_MOCKS}/>);

        expect(component.toJSON()).toMatchSnapshot();
    });

    it("It should display the message 'There is not skills available'", async () => {
        const component = create(<Projects skills={[]}/>);

        const div = component.root.findByProps({
            children: "There is not skills available"
        });

        expect(div.children).toEqual(["There is not skills available"]);
    });
});