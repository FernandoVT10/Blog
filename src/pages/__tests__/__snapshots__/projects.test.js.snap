// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project page It check if renders correctly 1`] = `
<div>
  <header>
    <nav
      className="navbar navbar-expand-md navbar-dark"
    >
      <a
        className="navbar-brand navbar__brand"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <img
          alt="Icon"
          className="d-inline-block align-top rounded-circle"
          height="50"
          src="/favicon.ico"
          width="50"
        />
        <h1
          className="navbar__title"
        >
          Fernando Vaca Tamayo
        </h1>
      </a>
      <button
        aria-controls="main-navbar"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-target="#main-navbar"
        data-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse navbar__items"
        id="main-navbar"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link navbar__link"
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link navbar__link"
              href="/articles"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Articles
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link navbar__link"
              href="/projects"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Projects
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link navbar__link"
              href="/contact-me"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Contact me
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </header>
  <div
    className="modal fade"
    id="projects-modal"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog modal-xl"
      role="document"
    >
      <div
        className="modal-content modal__content"
      >
        <div
          className="modal-header modal__header"
        >
          <h5
            className="modal__title"
          >
            undefined Projects
          </h5>
          <button
            aria-label="Close"
            className="close modal__button"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <div
          className="modal-body modal__body"
        >
          <div
            className="container-fluid p-0"
          >
            <div
              className="row"
            >
              <div
                className="col-12"
              >
                <h4
                  className="color-secondary text-center font-weight-bold"
                >
                  There is not available projects
                </h4>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="modal fade"
    id="project-modal"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog modal-lg"
      role="document"
    >
      <div
        className="modal-content modal__content"
      >
        <div
          className="modal-header modal__header"
        >
          <h5
            className="modal__title"
          >
            Loading...
          </h5>
          <button
            aria-label="Close"
            className="close modal__button"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <div
          className="modal-body modal__body"
        >
          <div
            className="container-fluid p-0"
          >
            <div
              className="row project-details"
            >
              <div
                className="loader loader--active"
                style={
                  Object {
                    "height": 300,
                  }
                }
              >
                <div
                  className="spinner-border text-light loader__spinner"
                  role="status"
                >
                  <span
                    className="sr-only"
                  >
                    Loading...
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="body"
  >
    <div
      className="container-fluid mt-4"
    >
      <div
        className="row projects-container"
      >
        <div
          className="col-12"
        >
          <h2
            className="title"
          >
            Skills
          </h2>
        </div>
        <div
          className="col-sm-6 col-md-6 col-lg-4 mb-3"
        >
          <div
            className="skill mb-3"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <div
              className="skill__image-container"
            >
              <img
                alt="React JS"
                className="skill__image"
                src="/img/skills/skill-1.jpg"
              />
              <span
                className="skill__image-hover"
                style={
                  Object {
                    "background": "blue",
                  }
                }
              />
            </div>
            <div
              className="skill__title"
              style={
                Object {
                  "background": "blue",
                }
              }
            >
              React JS
            </div>
          </div>
        </div>
        <div
          className="col-sm-6 col-md-6 col-lg-4 mb-3"
        >
          <div
            className="skill mb-3"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <div
              className="skill__image-container"
            >
              <img
                alt="Node JS"
                className="skill__image"
                src="/img/skills/skill-2.jpg"
              />
              <span
                className="skill__image-hover"
                style={
                  Object {
                    "background": "green",
                  }
                }
              />
            </div>
            <div
              className="skill__title"
              style={
                Object {
                  "background": "green",
                }
              }
            >
              Node JS
            </div>
          </div>
        </div>
        <div
          className="col-sm-6 col-md-6 col-lg-4 mb-3"
        >
          <div
            className="skill mb-3"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <div
              className="skill__image-container"
            >
              <img
                alt="GraphQl"
                className="skill__image"
                src="/img/skills/skill-3.jpg"
              />
              <span
                className="skill__image-hover"
                style={
                  Object {
                    "background": "purple",
                  }
                }
              />
            </div>
            <div
              className="skill__title"
              style={
                Object {
                  "background": "purple",
                }
              }
            >
              GraphQl
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="main-footer container-fluid mt-4"
  >
    <div
      className="full-screen-loader "
    >
      <div
        className="full-screen-loader__spinner spinner-border text-light"
        role="status"
      >
        <span
          className="sr-only"
        >
          Loading...
        </span>
      </div>
      <h3
        className="full-screen-loader__text"
      >
        Loading...
      </h3>
    </div>
    <div
      className="modal fade"
      id="-modal"
      role="dialog"
      tabIndex="-1"
    >
      <div
        className="modal-dialog "
        role="document"
      >
        <div
          className="modal-content modal__content"
        >
          <div
            className="modal-header modal__header"
          >
            <h5
              className="modal__title"
            >
              Suscription confirm status
            </h5>
            <button
              aria-label="Close"
              className="close modal__button"
              data-dismiss="modal"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </button>
          </div>
          <div
            className="modal-body modal__body"
          >
            <p>
              
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12 col-sm-6 col-xl-3"
      >
        <a
          className="main-footer__brand"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <img
            alt="Icon"
            className="d-inline-block align-top rounded-circle"
            height="50"
            src="/favicon.ico"
            width="50"
          />
          <h3
            className="main-footer__title"
          >
            Fernando Vaca Tamayo
          </h3>
        </a>
        <div
          className="mt-4"
        >
          <h3
            className="main-footer__subtitle"
          >
            Contact Me
          </h3>
          <p>
            <i
              aria-hidden="true"
              className="fas fa-envelope main-footer__icon"
            />
            <span
              className="main-footer__text"
            >
              vacatamayo321@gmail.com
            </span>
          </p>
        </div>
      </div>
      <div
        className="col-12 col-sm-6 col-xl-3"
      >
        <h3
          className="main-footer__subtitle"
        >
          Social Networks
        </h3>
        <a
          href="https://twitter.com/FernandoVT10"
          target="_blank"
        >
          <p>
            <i
              aria-hidden="true"
              className="fab fa-twitter main-footer__icon"
            />
            <span
              className="main-footer__text"
            >
              @FernandoVT10
            </span>
          </p>
        </a>
        <a
          href="https://github.com/FernandoVT10"
          target="_blank"
        >
          <p>
            <i
              aria-hidden="true"
              className="fab fa-github main-footer__icon"
            />
            <span
              className="main-footer__text"
            >
              FernandoVT10
            </span>
          </p>
        </a>
      </div>
      <div
        className="col-12 col-sm-6 col-xl-3"
      >
        <h3
          className="main-footer__subtitle"
        >
          Description
        </h3>
        <p
          className="main-footer__description"
        >
          This blog is my personal blog xD
        </p>
      </div>
      <div
        className="col-12 col-sm-6 col-xl-3"
      >
        <h3
          className="main-footer__subtitle"
        >
          Suscribe to my blog
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="formulary__input-container"
          >
            <input
              autoComplete="email"
              className="formulary__input "
              id="-input"
              maxLength="100"
              onChange={[Function]}
              placeholder="Enter your email"
              required={true}
              type="email"
            />
            <label
              className="formulary__error-label mt-2 "
              htmlFor="-input"
            >
              <i
                aria-hidden="true"
                className="fas fa-info-circle mr-1"
              />
              
            </label>
          </div>
          <button
            className="submit-button mt-3"
          >
            Suscribe
          </button>
        </form>
      </div>
    </div>
  </footer>
  <script
    src="https://code.jquery.com/jquery-3.4.1.slim.min.js"
  />
  <script
    src="https://cdn.jsdelivr.net/npm/popper.js@1.16.0/dist/umd/popper.min.js"
  />
  <script
    src="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/js/bootstrap.min.js"
  />
  <script
    src="https://kit.fontawesome.com/63ef8f1397.js"
  />
</div>
`;
